### PR TITLE
Create one AWS VPC/IGW/PlacementGroup per region.

### PR DESCRIPTION
**Pull request by cmccoy**
VPCs, Internet Gateways, and Placement groups are regional resources.
This change creates a single VPC/IGW/Placement Group per region (via
`AwsRegionalNetwork`), then creates a per-AZ subnet.

Tested with:

    python pkb.py --benchmarks netperf --cloud AWS \
      --zones us-west-1a,us-west-1c \
      --ip_addresses INTERNAL \
      --netperf_test_length 10

    python pkb.py --benchmarks netperf --cloud AWS \
      --zones us-west-1 \
      --ip_addresses INTERNAL \
      --netperf_test_length 10

### CODE REVIEW

**Comment by cmccoy** [comment]
@ehankland - would you mind taking a look?

File: perfkitbenchmarker/providers/aws/aws_network.py [hunk 1]
```diff
@@ -361,15 +385,103 @@ def _Exists(self):
...
+  """Object representing regional components of an AWS network.
+
+  This class maintains a singleton-per-region; acquire instances via
+  AwsRegionalNetwork.GetForRegion.
+
+  Attributes:
+    region: string. The AWS region.
+    vpc: an AwsVpc instance.
+    internet_gateway: an AwsInternetGateway instance.
+    route_table: The default route table.
```
Comment (yuyantingzero) [inline perfkitbenchmarker/providers/aws/aws_network.py]: Type of a route_table?

File: perfkitbenchmarker/providers/aws/aws_network.py [hunk 2]
```diff
@@ -139,6 +145,24 @@ def _Delete(self):
...
 
+  def NextSubnetCidrBlock(self):
+    """Returns the next available /24 CIDR block in this VPC.
+
+    Each VPC has a 10.0.0.0/16 CIDR block.
+    Each subnet is assigned a /24 within this allocation.
+    Calls to this method return the next unused /24.
+
+    Returns:
+      A /24 CIDR block.
```
Comment (yuyantingzero) [inline perfkitbenchmarker/providers/aws/aws_network.py]: clearify the return type.

File: perfkitbenchmarker/providers/aws/aws_network.py [hunk 3]
```diff
@@ -139,6 +145,24 @@ def _Delete(self):
         '--vpc-id=%s' % self.id]
     vm_util.IssueCommand(delete_cmd)
 
+  def NextSubnetCidrBlock(self):
+    """Returns the next available /24 CIDR block in this VPC.
+
+    Each VPC has a 10.0.0.0/16 CIDR block.
+    Each subnet is assigned a /24 within this allocation.
+    Calls to this method return the next unused /24.
+
```
Comment (yuyantingzero) [inline perfkitbenchmarker/providers/aws/aws_network.py]: Raises section?

Comment (cmccoy) [inline perfkitbenchmarker/providers/aws/aws_network.py]: Done.

File: perfkitbenchmarker/providers/aws/aws_network.py (hunk 2 above)
Comment (cmccoy) [inline perfkitbenchmarker/providers/aws/aws_network.py]: Done.

File: perfkitbenchmarker/providers/aws/aws_network.py (hunk 1 above)
Comment (cmccoy) [inline perfkitbenchmarker/providers/aws/aws_network.py]: Done.

**Comment by yuyantingzero** [comment]
LGTM

**Pull request merged**